### ISSUE TITLE
python3 train_condition.py forward() missing 1 required positional argument: 'input2'

### ISSUE THREAD

**Issue by OliverVarnce** [sangyun884/HR-VITON#50]
Can you help me with train_condition.py?

I have an error:

File "train_condition.py", line 499, in <module>
    main()
  File "train_condition.py", line 490, in main
    train(opt, train_loader, val_loader, test_loader, board, tocg, D)
  File "train_condition.py", line 159, in train
    flow_list, fake_segmap, warped_cloth_paired, warped_clothmask_paired = tocg(input1, input2)
  File "/home/user/.local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
TypeError: forward() missing 1 required positional argument: 'input2'

What is the problem?

**Comment by itouchzh** [comment]
you can test add a parameter for the model? example this images, I can solve this problem in this way
![image](https://user-images.githubusercontent.com/86830424/213896514-1ebabc20-9eb0-4d1f-ab41-926ddc81c468.png)

**Comment by OliverVarnce** [comment]
Thanks for response. I will test this idea ASAP.

**Comment by OliverVarnce** [comment]
I have this error on 100th step. Whats may be wrong?

**Comment by OliverVarnce** [comment]
> you can test add a parameter for the model? example this images, I can solve this problem in this way ![image](https://user-images.githubusercontent.com/86830424/213896514-1ebabc20-9eb0-4d1f-ab41-926ddc81c468.png)

I have this error on 100th step. Whats may be wrong?

**Comment by HyunseokDE** [comment]
Did you solve this problem?

**Comment by OliverVarnce** [comment]
> Did you solve this problem?

yes. Inc code we have  ... = tocg(input1, input2) but should be ... = tocg(opt, input1, input2)

**Comment by HyunseokDE** [comment]
> > Did you solve this problem?
> 
> yes. Inc code we have ... = tocg(input1, input2) but should be ... = tocg(opt, input1, input2)
thank you,
hmm problem is persists..I need to find more

**Comment by Chae-Mingi** [comment]
Did you solve this problem?

**Comment by OliverVarnce** [comment]
> Did you solve this problem?

Yes. Adding 'opt' in first parameter in call tocg solves this problem - tocg(opt, input1, input2)

**Comment by 111wukong** [comment]
Thank you, but the method you provided did not solve the problem.

**Comment by SXJCLZH** [comment]
Big guy, I tried to run it again according to the method you said, but it still doesn't work. Could you please describe it in more detail or share your key code?
`
TypeError: make_grid() missing 1 required positional argument: 'opt'
`

**Comment by ntad27** [comment]
could anyone please describe more detail how to fix it?

**Comment by whateverisnottaken** [comment]
In train_condition.py, 
change line 158 from `...tocg(input1, input2)` to `...tocg(opt, input1, input2)`
change line 481 from `...ConditionGenerator(opt...` to `...ConditionGenerator(opt=opt...`
change line 241 from `...mkgrid(N, iH, iW)` to `...mkgrid(N, iH, iW, opt)`
solves the problem on my machine.
